webhookValidationConfig.go

```diff
@@ -4,47 +4,31 @@
 package ancla
 
 import (
-	"errors"
-	"fmt"
 	"time"
-)
 
-var (
-	SpecialUseIPs = []string{
-		"0.0.0.0/8",          //local ipv4
-		"fe80::/10",          //local ipv6
-		"255.255.255.255/32", //broadcast to neighbors
-		"2001::/32",          //ipv6 TEREDO prefix
-		"2001:5::/32",        //EID space for lisp
-		"2002::/16",          //ipv6 6to4
-		"fc00::/7",           //ipv6 unique local
-		"192.0.0.0/24",       //ipv4 IANA
-		"2001:0000::/23",     //ipv6 IANA
-		"224.0.0.1/32",       //ipv4 multicast
-	}
-	SpecialUseHosts = []string{
-		".example.",
-		".invalid.",
-		".test.",
-		"localhost",
-	}
-	errFailedToBuildValidators    = errors.New("failed to build validators")
-	errFailedToBuildValidURLFuncs = errors.New("failed to build ValidURLFuncs")
+	"github.com/xmidt-org/urlegit"
 )
 
 type ValidatorConfig struct {
-	URL URLVConfig
-	TTL TTLVConfig
+	URL    URLVConfig
+	TTL    TTLVConfig
+	IP     IPConfig
+	Domain DomainConfig
+}
+
+type IPConfig struct {
+	Allow            bool
+	ForbiddenSubnets []string
+}
+
+type DomainConfig struct {
+	AllowSpecialUseDomains bool
+	ForbiddenDomains       []string
 }
 
 type URLVConfig struct {
-	HTTPSOnly            bool
-	AllowLoopback        bool
-	AllowIP              bool
-	AllowSpecialUseHosts bool
-	AllowSpecialUseIPs   bool
-	InvalidHosts         []string
-	InvalidSubnets       []string
```

Comment:
do we want to remove these three configurations?

Reply:
that's correct. this refactor was from Wes after I had my initial conversation with him last week when I had questions about  the urlegit package.